tests/unit/test_templatevars.py

```diff
@@ -141,3 +149,10 @@ def test_get_old_tvars(key, expect, _setup_db):
     """It can extract a variety of items from a workflow database.
     """
     assert _setup_db[key] == expect
+
+
+def test_dont_get_old_tvars_back_compat(tmp_path):
+    """It won't even try to extract workflow_template_vars in compat mode.
+    """
+    tmp_path.touch('suite.rc')
+    assert get_template_vars_from_db(tmp_path) == {}
```

Comment:
I don't think this test is valid - why shouldn't it extract template vars in back compat mode?

However, I think we should test that parsing the config fails if a Cylc 7 DB is present

Reply:
This will result in an SQL error (table was renamed).

Reply:
I have now added a test where a completely blank file is checked.